### ISSUE TITLE
2.11.0 ubuntu - error writing data from tail - log filled up the disk 

### ISSUE THREAD

**Issue by rubber-ant** [GoogleCloudPlatform/ops-agent#463]
Ref https://github.com/fluent/fluent-bit/issues/4766

Happened the same on  Ubuntu 20.04.4, with 2 VMs 

example of one of them:

`google-cloud-ops-agent `version `2.11.0~ubuntu20.04`

huge file log was created: 
```
ls -alh /var/log/google-cloud-ops-agent/subagents/logging-module.log
-rw-r--r-- 1 root root 117G Mar  7 14:52 /var/log/google-cloud-ops-agent/subagents/logging-module.log
```

<img width="1540" alt="image" src="https://user-images.githubusercontent.com/2983049/158071931-a3071c1e-282c-423f-b494-2ec91a68804a.png">

Inside the log file:
```
[2022/03/05 11:48:03] [error] [storage] format check failed: tail.2/870-1646420538.862293556.flb
[2022/03/05 11:48:03] [error] [storage] format check failed: tail.2/870-1646420538.862293556.flb
[2022/03/05 11:48:03] [error] [storage] [cio file] file is not mmap()ed: tail.2:870-1646420538.862293556.flb
[2022/03/05 11:48:03] [error] [input chunk] error writing data from tail.2 instance
[2022/03/05 11:48:03] [ info] [task] re-schedule retry=0x7fea501439e0 77 in the next 7 seconds
[2022/03/05 11:48:03] [ info] [task] re-schedule retry=0x7fea500b2160 51 in the next 11 seconds
[2022/03/05 11:48:04] [ info] [task] re-schedule retry=0x7fea501538d0 79 in the next 7 seconds
[2022/03/05 11:48:04] [ info] [task] re-schedule retry=0x7fea50150900 84 in the next 21 seconds
[2022/03/05 11:48:04] [ info] [task] re-schedule retry=0x7fea50146900 33 in the next 15 seconds
[2022/03/05 11:48:04] [error] [storage] format check failed: tail.2/870-1646420538.862293556.flb
[2022/03/05 11:48:04] [error] [storage] format check failed: tail.2/870-1646420538.862293556.flb
[2022/03/05 11:48:04] [error] [storage] [cio file] file is not mmap()ed: tail.2:870-1646420538.862293556.flb
[2022/03/05 11:48:04] [error] [input chunk] error writing data from tail.2 instance
[2022/03/05 11:48:04] [ info] [task] re-schedule retry=0x7fea5013f9c0 76 in the next 19 seconds
[2022/03/05 11:48:04] [error] [storage] format check failed: tail.2/870-1646420538.862293556.flb
[2022/03/05 11:48:04] [error] [storage] format check failed: tail.2/870-1646420538.862293556.flb
[2022/03/05 11:48:04] [error] [storage] [cio file] file is not mmap()ed: tail.2:870-1646420538.862293556.flb
[2022/03/05 11:48:04] [error] [input chunk] error writing data from tail.2 instance
[2022/03/05 11:48:05] [error] [storage] format check failed: tail.1/870-1646420636.219474538.flb
[2022/03/05 11:48:05] [error] [storage] format check failed: tail.1/870-1646420636.219474538.flb
[2022/03/05 11:48:05] [error] [storage] [cio file] file is not mmap()ed: tail.1:870-1646420636.219474538.flb
[2022/03/05 11:48:05] [error] [input chunk] error writing data from tail.1 instance
```

Same scenario with the second VM, disk filled up with this logs in 7 hours:
<img width="1570" alt="image" src="https://user-images.githubusercontent.com/2983049/158071951-aef5ecd5-6306-45ea-828b-30c44477ef97.png">

Any suggestions?

**Comment by sophieyfang** [comment]
Can you try below commands, this is to clean up the old states:

sudo service google-cloud-ops-agent stop && 
sudo rm -f /var/log/google-cloud-ops-agent/subagents/logging-module.log && 
sudo rm -rf /var/lib/google-cloud-ops-agent/fluent-bit/buffers/*/ &&
sudo service google-cloud-ops-agent restart

**Comment by rubber-ant** [comment]
@sophieyfang I followed your steps in all VMS, unfortunately, it still happening today!

```
sudo service google-cloud-ops-agent stop &&
sudo rm -f /var/log/google-cloud-ops-agent/subagents/logging-module.log &&
sudo rm -rf /var/lib/google-cloud-ops-agent/fluent-bit/buffers/*/ &&
sudo service google-cloud-ops-agent restart
```

```
$ date; ls -alh /var/log/google-cloud-ops-agent/subagents/logging-module.log
Thu Mar 31 09:09:58 BST 2022
-rw-r--r-- 1 root root 63G Mar 31 09:09 /var/log/google-cloud-ops-agent/subagents/logging-module.log
```
```
ls -alh /var/lib/google-cloud-ops-agent/fluent-bit/buffers/
total 8.0M
drwxr-xr-x 4 root root 4.0K Mar 30 08:27 .
drwxr-xr-x 3 root root 4.0K Feb 25 16:58 ..
-rw-r--r-- 1 root root 8.0K Mar 30 12:30 default_pipeline_syslog
-rw-r--r-- 1 root root  32K Mar 31 01:46 default_pipeline_syslog-shm
-rw-r--r-- 1 root root 4.0M Mar 31 01:46 default_pipeline_syslog-wal
-rw-r--r-- 1 root root 8.0K Mar 31 09:17 ops-agent-fluent-bit
-rw-r--r-- 1 root root  32K Mar 31 09:17 ops-agent-fluent-bit-shm
-rw-r--r-- 1 root root 4.0M Mar 31 09:17 ops-agent-fluent-bit-wal
drwxr-xr-x 2 root root 4.0K Mar 30 21:06 tail.1
drwxr-xr-x 2 root root 4.0K Mar 30 20:02 tail.2
```

```
dpkg -l | grep google-cloud-ops-agent
ii  google-cloud-ops-agent               2.11.0~ubuntu20.04                amd64        Google Cloud Ops Agent
```


<img width="449" alt="image" src="https://user-images.githubusercontent.com/2983049/161009734-bf5696c0-cbea-46aa-bf92-ed804cbabaa5.png">


 Noticed new versions: 
```
google-cloud-ops-agent | 2.12.1~ubuntu20.04 | https://packages.cloud.google.com/apt google-cloud-ops-agent-focal-all/main amd64 Packages
google-cloud-ops-agent | 2.12.0~ubuntu20.04 | https://packages.cloud.google.com/apt google-cloud-ops-agent-focal-all/main amd64 Packages
```

before upgrading the version from 2.11.0 to 2.12.1, do I need to clean the following?
```
sudo rm -f /var/log/google-cloud-ops-agent/subagents/logging-module.log &&
sudo rm -rf /var/lib/google-cloud-ops-agent/fluent-bit/buffers/*/ &&
```

**Comment by rubber-ant** [comment]
@sophieyfang it seems that it started to write logs when memory was released, I remember the same scenario for the other VMS when the same issue occurred.

<img width="774" alt="image" src="https://user-images.githubusercontent.com/2983049/161025233-78d80ead-9458-43d9-a01f-c6cd8d600b29.png">

**Comment by sophieyfang** [comment]
```
sudo service google-cloud-ops-agent stop &&
sudo rm -f /var/log/google-cloud-ops-agent/subagents/logging-module.log &&
sudo rm -rf /var/lib/google-cloud-ops-agent/fluent-bit/buffers/*/ &&
sudo service google-cloud-ops-agent restart
```

Did you verify there were no files under `/var/lib/google-cloud-ops-agent/fluent-bit/buffers/*/` before you restarted the agent? I wonder if it was still the old buffer files lingering somewhere.
Also, if you could, to better assist you, you can raise a google cloud support case that way we have more visibility into your problem 😊

**Comment by Tassosb** [comment]
We encountered the same issue on multiple of our centos 7 VMs.
Ops Agent Version: 2.12.0
We were using the legacy logging agent prior to installing Ops Agent 2.12 in March 2022, so I don't see how there could have been old buffer files in `/var/lib/google-cloud-ops-agent/fluent-bit/buffers`.

We performed the commands suggest in the message above to fix the issue on our affected VMs. Can we expect that to fix the issue long-term? I'm worried about this issue cropping up on our other VMs, on which we installed Ops Agent 2.12 in the exact same way.
